cmake/TorchConfig.cmake.in

```diff
@@ -13,6 +13,27 @@
 # and the following imported targets:
 #
 #   torch
+macro(append_torchlib_if_found)
+  foreach (_arg ${ARGN})
+    find_library(${_arg}_LIBRARY ${_arg} PATHS "${TORCH_INSTALL_PREFIX}/lib")
+    if(${_arg}_LIBRARY)
+      list(APPEND TORCH_LIBRARIES ${${_arg}_LIBRARY})
+    else()
+      message(WARNING "static library ${${_arg}_LIBRARY} not found.")
```

Comment:
Can you explain why this one should not be an error?
Also, why this macro is specific to a static libraries discovery?

Reply:
warning instead of error mainly because of two reasons:
1, I cannot find the corresponding CMake option for every individual static library, so if the static library is not found, means the implied cmake option is off (rather than an error);
2, some static libraries are not mandatory for end user, warning instead of error provides flexibility in this situation.

Reply:
Hmm, can you give an example of optional library dependencies?
Also "static library" message is misleading, as `append_torchlib_if_found` is called from both shared and static library path.

Reply:
eigen_blas is optional, as we may use MKL as lapack backend, and I cannot find a public cmake option to control this,e.g.:
```
if(@USE_EIGENBLAS@)
```
If eigen_blas option can be found, and also caffe2_protos, protobuf-lite, protobuf, protoc, onnx, onnx_proto, foxi_loader, fmt, sleef, asmjit are all mandatory library for static build, we can change the warning to error.